KIC: feat: add introduction about combined service from different httproutes (#8270)
Kong/docs.konghq.com: app/_src/kubernetes-ingress-controller/reference/feature-gates.md

> **Warning:** `KongPlugin`'s and `KongClusterPlugin`'s `config` fields are not sanitized. If you have sensitive information
> in your `KongPlugin`'s `config` field, it will be sent to Konnect as is. To avoid that, please consider using
> [KongVault](/kubernetes-ingress-controller/{{page.release}}/reference/custom-resources/#kongvault).

> ### CombinedServicesFromDifferentHTTPRoutes

Reviewer: This needs version conditions no?

Author: Added version conditions here.

> The `CombinedServicesFromDifferentHTTPRoutes` feature enables translating `HTTPRoute` rules

Reviewer: ```suggestion
The `CombinedServicesFromDifferentHTTPRoutes` feature enables translating `HTTPRoute`  in the same namespace
```

> When the feature gate `CombinedServicesFromDifferentHTTPRoutes` is enabled, {{site.kic_product_name}} will translate `HTTPRoute` rules

Reviewer: Can we preserve a syntax similar to the previous one?

```suggestion
The `CombinedServicesFromDifferentHTTPRoutes` feature enables the sharing of the same set of backends `HTTPRoute` rules...
```

> having the same set of backends (combination of namespace, name, port and weight) from different `HTTPRoute`s in the same namespace

Reviewer: ```suggestion
with the same set of backends
```

Author: backends belong to a rule in `HTTPRoute` but not the whole `HTTPRoute`. So I think the doc should mention that the consolidation is applied on rules of `HTTPRoute`s.

> sharing the same set of backends(comblination of namespace, name, port and weight) from different `HTTPRoute`s in the same namespace

Reviewer 2: ```suggestion
sharing the same set of backends (combination of namespace, name, port and weight) from different `HTTPRoute`s in the same namespace
```

> into one {{site.base_gateway}} service. Enabling the feature gate could reduce the number of translated {{site.base_gateway}} services.

Reviewer: ```suggestion
into a single {{site.base_gateway}} service. Enabling the feature gate can reduce the number of {{site.base_gateway}} services.
```

Author: done.

> The names of {{site.base_gateway}} services will be changed if the feature gate is enabled.

Reviewer: ```suggestion
The names of {{site.base_gateway}} services will change if the feature gate is enabled.
```

Author: Updated.

> The names of {{site.base_gateway}} services will be changed if the feature gate is enabled compared to the names when it is disabled.

Reviewer: ```suggestion
The names of {{site.base_gateway}} services will be changed if the feature gate is enabled.
```

No need for the extra verbosity here

Author: Done.

> You can refer to the [reference page](/kubernetes-ingress-controller/{{page.release}}/reference/combined-services-from-different-httproutes) to know the details.

Reviewer: ```suggestion
You can refer to the [reference page](/kubernetes-ingress-controller/{{page.release}}/reference/combined-services-from-different-httproutes) for further details.
```

Author: done.
